packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -69,6 +108,22 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             Lib_ExecutionManagerWrapper.ovmCHAINID()
         );
 
+        // Compute the intrinsic gas
+        uint256 intrinsicGas = _intrinsicGas(
+            _encodedTransaction,
+            transaction.isCreate
+        );
+
+        // Decode the L2 gas limit. It is scaled and serialized in the lower
+        // order bits of transaction.gasLimit. See:
+        // https://github.com/ethereum-optimism/optimism/blob/0c18e1903f8a33a607f782c0081a8fb5071b71ef/l2geth/rollup/fees/rollup_fee.go#L56
+        uint256 gasLimit = SafeMath.mul((transaction.gasLimit % 10000), 10000);
+        // Need to make sure that the gas is sufficient to execute the transaction.
```

Comment:
If I understand this correctly, the whole point of this require statement is to prevent the case in which the transaction runs out of gas before the nonce is incremented. I am not sure if `intrinsicGas` is the correct term we are looking for (that to me implies the gas used from the beginning of the sequencer entrypoint up to the ecdsa contract account sub call) but I do believe that we want this require statement to guarantee there is enough gas to increment the nonce, meaning the the `gasleft` is larger than the L2 gas limit + the amount of gas consumed getting to the ecdsa contract account call.

This means that the `intrinsicGas` function should return the amount of gas that `execute` uses up to `transaction.to.call` or the returns after `Lib_ExecutionManagerWrapper.ovmCREATE` in the case of a contract creation.

The full L2 gas limit is then passed into the subcalls

Reply:
I would be comfortable calling this `EXECUTE_INTRINSIC_GAS` to reflect that this is just for the `OVM_ECDSAContractAccount.execute` function and not the sequencer entrypoint, proxyEOA, etc.